src/sentry/profiles/task.py

```diff
@@ -221,18 +221,26 @@ def _process_symbolicator_results_for_sample(profile: Profile, stacktraces: List
             frame.pop("context_line", None)
             frame.pop("post_context", None)
 
-        def truncate_stack_needed(frame: dict[str, Any]) -> bool:
-            return bool(frame.get("function", "") == "perf_signal_handler")
+        def truncate_stack_needed(
+            frame: dict[str, Any], stack: List[Any]
+        ) -> Tuple[bool, List[Any]]:
+            # remove top frames related to the profiler
+            return bool(frame.get("function", "") == "perf_signal_handler"), stack[2:]
```

Comment:
Shouldnt this be
```suggestion
            return bool(frame.get("function", "") == "perf_signal_handler"), stack[:-2]
```
so that we drop the last 2 frames?

Reply:
I checked with @viglia and it turns out for Rust, we want to drop the first 2 frames as `perf_signal_handler` is indeed called last. Stack is still bottom to top like for `cocoa` but what we want to drop is at the bottom of the stack.